File: CombinationSumIII.cpp
class Solution {
public:
    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> result;
        vector<int> oneCase;
        recursive(1,k,n,oneCase,result);
        return result;
    }
    
    void recursive(int start, int k, int n, vector<int> oneCase, vector<vector<int>>& result){
        if(start > 9 || k < 1){
            return;
        }
        for(int i = start; i < 10; i++){
            if(i > n){
                return;
            }else if(i == n){
                if(k == 1){
                    oneCase.push_back(i);
                    vector<int> final(oneCase);
                    result.push_back(final);
                    oneCase.pop_back();
                }else{
                    return;
                }
            }else{
                oneCase.push_back(i);
                recursive(i+1,k-1,n - i, oneCase,result); 
                oneCase.pop_back();
            }
        }
    }
};

File: python_src/CombinationSumIII.py
class Solution(object):
    def combinationSum3(self, k, n):
        """
        :type k: int
        :type n: int
        :rtype: List[List[int]]
        """
        result = []
        self.getCombination(result, [], k, n, 1)
        return result
    
    def getCombination(self,result, one_case, k, n, start):
        if 0 == k and n == 0:
            result.append(list(one_case))
            return
        
        for i in range(start,10):
            if i > n: return
            one_case.append(i)
            self.getCombination(result, one_case, k - 1, n - i, i + 1)
            del one_case[-1]
        
            